### ISSUE TITLE
Custom controller and access to properties

### ISSUE THREAD

**Issue by ambeam** [sulu/sulu#2366]
Hey there, I'm not sure where to ask, so I decided to write here. Could anyone instruct me how to use properties defined in page XML in twig template from custom controller?

For example, here's my XML:

    <key>staticPage</key>
    <view>ClientWebsiteBundle:templates:default</view>
    <controller>ClientWebsiteBundle:Static:view</controller>
    <cacheLifetime>2400</cacheLifetime>
    <properties>
        ...
        <property name="page_content" type="text_editor">
            <meta>
                <title lang="de">Content</title>
                <title lang="en">Content</title>
                <title lang="pl">Treść</title>
            </meta>
        </property>
        ...

In my twig template I have no access to `page_content`, the same is with page title from `<property name="title" type="text_line" mandatory="true">` property. Docs says that I should use `{{ content.title }}` to echo page title, but it's accessible from default controller only (`SuluWebsiteBundle:Default:index`).

I think the documentation seems to be a bit unclear for newbie users, even for those who has some experience with Sf2. Beside of that, I'd like to work more with Sulu, that's why I'm asking for help.

Thanks in advance.

**Comment by wachterjohannes** [comment]
hello @ambeam the right place would be our public slack channel ask for access on our homepage http://sulu.io/en#questions

now to your question (: can you show me how you defined your own controller `ClientWebsiteBundle:Static:view `? the easiest way to define this controller is to inherit from our `Sulu\Bundle\WebsiteBundle\Controller\WebsiteController` and add a function like this:

```php
    /**
     * Loads the content from the request (filled by the route provider) and creates a response with this content and
     * the appropriate cache headers.
     *
     * @param \Sulu\Component\Content\Compat\StructureInterface $structure
     * @param bool $preview
     * @param bool $partial
     *
     * @return Response
     */
    public function indexAction(StructureInterface $structure, $preview = false, $partial = false)
    {
        $response = $this->renderStructure(
            $structure,
            [],
            $preview,
            $partial
        );

        return $response;
    }
```

**Comment by ambeam** [comment]
Hey @wachterjohannes , thanks for your reply. I've already logged in to the slack channel.

I've created a very simple action in my controller, that returns rendered twig template only (so far). However I've tried the method You mentioned (from `WebsiteController`) but I had no idea how to pass my own twig template there. `renderStructure` method calls `getView()` so I thought there is some `setView()` method in `StructureInterface` but there's no any like this. What I am doing wrong? :)
